### PR TITLE
JERSEY-2891: ExceptionMapper not chosen correctly in case of multiple ExtendedExceptionMappers with same generic type

### PR DESCRIPTION
**Pull request by kppk**
Current implementation ExceptionMapperFactory#find method used sorted map to store the mappers by distance of the generic exception type from the actual type. If there were two exception mappers with the same generic type, the mappers got overwritten in that map. Also I noticed that the exception mappers order was changing with server restart making the behavior unpredictable. 

This fix removed the sorted map, finds the mapper with the lowest distance doing the 'isMappable' check within the same loop.

### CODE REVIEW

**Comment by mpotociar** [comment]
Jenkins, please test this patch.

File: core-common/src/test/java/org/glassfish/jersey/internal/ExceptionMapperFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,550 @@
+package org.glassfish.jersey.internal;
```
Comment (mpotociar) [inline core-common/src/test/java/org/glassfish/jersey/internal/ExceptionMapperFactoryTest.java]: Missing copyright header. Please use a header e.g. from here: https://github.com/jersey/jersey/blob/master/core-common/src/test/java/org/glassfish/jersey/process/internal/ExecutorProvidersTest.java

File: core-common/src/test/java/org/glassfish/jersey/internal/ExceptionMapperFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+            return Objects.hash(super.hashCode(), impl);
+        }
+    }
+
+    /**
+     * Stub implementation -- reason for this is to control the order of list of services in ServiceLocator.
+     *
+     * @param <T>
+     */
+    private static class ServiceHandleStub<T> implements ServiceHandle<T> {
```
Comment (mpotociar) [inline core-common/src/test/java/org/glassfish/jersey/internal/ExceptionMapperFactoryTest.java]: I guess that better than mocking the locator, you can just configure one for testing purposes, e.g. like here:
https://github.com/jersey/jersey/blob/master/core-common/src/test/java/org/glassfish/jersey/process/internal/ExecutorProvidersTest.java#L238-L256

To simulate user-registered exception mappers (if needed), you can create bindings using the "@Custom" qualifier:
https://github.com/jersey/jersey/blob/master/core-common/src/main/java/org/glassfish/jersey/internal/inject/Custom.java

**Comment by mgajdos** [comment]
Verified manually.

**Comment by mgajdos** [comment]
Thanks for your contribution.

**Pull request merged**